### PR TITLE
Draft PR supporting geneExpression

### PR DESCRIPTION
**Pull request by airenzp**
Description

This PR adds support to geneExpression, and as it is on top of the geneVariant PR, it also supports the changes in this type, like using coordinates to search. It also does a major refactor to solve the problems related to accessing sample term values. It can be tested using the [TermdbTest](http://localhost:3000/?noheader=1&mass={%22genome%22:%22hg19%22,%22dslabel%22:%22panallMds3%22}](http://localhost:3000/?noheader=1&mass={%22genome%22:%22hg38-test%22,%22dslabel%22:%22TermdbTest%22})) dataset. To test the geneExpression term, from the scatter plot, do overlay and select the gene expression tab, then select  a gene expression. You would see something like this:

Screenshot 2024-04-17 at 10 08 53 PM

We also require further testing of all the functionalities, as the refactoring updated the way the data is returned by the getData method, which is central to most plots. So please check everything works as expected.

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [x] Tests: added and/or passed unit and integration tests, or N/A
- [x] Todos: commented or documented, or N/A
- [x] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

File: client/termsetting/handlers/geneExpression.ts [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+/* 
+instance attributes
+
+self.term{}
+	.name: str, not really used
+	.type: "geneExpression"
+*/
+
+//TODO move to common.ts??? Corresponds to client/shared/common.js
+type MClassEntry = { label: string; color: string; dt: number; desc: string; key: string }
```
Comment (xzhou82) [inline client/termsetting/handlers/geneExpression.ts:14]: mclass is for mutation. can delete these for gene expression handler

File: server/shared/types/terms/geneExpression.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	mode: 'continuous'
+}
+
+export type GeneExpressionTW = TermWrapper & {
+	q: GeneExpressionQ
+	term: GeneExpressionTerm
+}
+
+export type GeneExpressionTerm = Term & {
+	gene?: string
```
Comment (xzhou82) [inline server/shared/types/terms/geneExpression.ts:28]: change to required

File: server/src/termdb.matrix.js [hunk 3]
```diff
@@ -147,6 +148,26 @@ async function getSampleData(q) {
...
+				genome: q.ds.genome,
+				dslabel: q.ds.label,
+				clusterMethod: 'hierarchical',
+				/** distance method */
+				distanceMethod: 'euclidean',
+				/** Data type */
+				dataType: 3,
+				genes: [{ gene: tw.term.gene }]
+			}
+			const data = await getGeneExpression(args)
```
Comment (xzhou82) [inline server/src/termdb.matrix.js:163]: no need

File: server/src/termdb.matrix.js [hunk 4]
```diff
@@ -147,6 +148,26 @@ async function getSampleData(q) {
 
 				samples[sampleId][tw.term.id] = snp2value
 			}
+		} else if (tw.term.type == TermTypes.GENE_EXPRESSION) {
+			const getGeneExpression = q.ds.queries.geneExpression.get
```
Comment (xzhou82) [inline server/src/termdb.matrix.js:152]: no need to define a new variable. directly call get()

**Review by gavrielm — APPROVED** [review]
We will merge this branch into the `geneVariant-refactor` branch and continue working off of that branch.

**Pull request merged**